internal/resources/policy/kind/network/input_schema.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package policykindnetwork
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	policyrecipenetworkmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/network"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy"
+	reciperesource "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy/kind/network/recipe"
+)
+
+var (
+	inputSchema = &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Input for the network policy, having one of the valid recipes: allow-all, allow-all-to-pods, allow-all-egress, deny-all, deny-all-to-pods, deny-all-egress, custom-egress or custom-ingress.",
+		Required:    true,
+		MaxItems:    1,
+		MinItems:    1,
+		ForceNew:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				reciperesource.AllowAllKey:       reciperesource.AllowAll,
+				reciperesource.AllowAllToPodsKey: reciperesource.AllowAllToPods,
+				reciperesource.AllowAllEgressKey: reciperesource.AllowAllEgress,
+				reciperesource.DenyAllKey:        reciperesource.DenyAll,
+				reciperesource.DenyAllToPodsKey:  reciperesource.DenyAllToPods,
+				reciperesource.DenyAllEgressKey:  reciperesource.DenyAllEgress,
+				reciperesource.CustomEgressKey:   reciperesource.CustomEgress,
+				reciperesource.CustomIngressKey:  reciperesource.CustomIngress,
+			},
+		},
+	}
+	RecipesAllowed = [...]string{reciperesource.AllowAllKey, reciperesource.AllowAllToPodsKey, reciperesource.AllowAllEgressKey, reciperesource.DenyAllKey, reciperesource.DenyAllToPodsKey, reciperesource.DenyAllEgressKey, reciperesource.CustomEgressKey, reciperesource.CustomIngressKey}
+)
+
+type (
+	Recipe string
+	// InputRecipe is a struct for all types of network policy inputs.
+	inputRecipe struct {
+		recipe              Recipe
+		inputAllowAll       *policyrecipenetworkmodel.V1alpha1CommonPolicySpecNetworkV1AllowAll
```

Comment:
`recipe` is being used for all recipes to store the recipe name. The 3 recipes that don't have an input struct is because they don't have a defined model and no specific schema. I had shown this in the demo.

Reply:
Why should Recipe be global ?

Reply:
Changed to local